cpp/vm/evmzero/interpreter_test.cc

```diff
@@ -4448,6 +4448,84 @@ TEST(InterpreterTest, JUMPDEST_OutOfGas) {
   });
 }
 
+///////////////////////////////////////////////////////////
+// TLOAD
+TEST(InterpreterTest, TLOAD) {
+  MockHost host;
+  EXPECT_CALL(host, get_transient_storage(evmc::address(0x42), evmc::bytes32(16)))  //
+      .Times(1)
+      .WillOnce(Return(evmc::bytes32(32)));
+
+  RunInterpreterTest({
+      .code = {op::TLOAD},
+      .state_after = RunState::kDone,
+      .gas_before = 100,
+      .gas_after = 0,
+      .stack_before = {16},
+      .stack_after = {32},
+      .message = {.recipient = evmc::address(0x42)},
+      .host = &host,
+      .revision = EVMC_CANCUN,
+  });
+}
+
+TEST(InterpreterTest, TLOAD_OutOfGas) {
+  RunInterpreterTest({
+      .code = {op::TLOAD},
+      .state_after = RunState::kErrorGas,
+      .gas_before = 99,
+      .stack_before = {0},
+      .revision = EVMC_CANCUN,
+  });
+}
+
+TEST(InterpreterTest, TLOAD_PreCancun) {
+  RunInterpreterTest({
+      .code = {op::TLOAD},
+      .state_after = RunState::kErrorOpcode,
+      .gas_before = 100,
+      .revision = EVMC_PARIS,
+  });
+}
+
+///////////////////////////////////////////////////////////
+// TSTORE
```

Comment:
stack underflow tests ?

Reply:
Since stack underflows are not handled inside the instruction implementation, I don't think it should be tested here.